projects/addon-table/components/table/directives/table.directive.ts

```diff
@@ -63,21 +63,30 @@ export class TuiTableDirective<
     @tuiDefaultProp()
     sorter: TuiComparator<T> = () => 0;
 
-    updateSorter(sorter: TuiComparator<T> | null): void {
+    updateSorterAndDirection(sorter: TuiComparator<T> | null): void {
         if (this.sorter === sorter) {
-            this.direction = this.direction === 1 ? -1 : 1;
-            this.directionChange.emit(this.direction);
+            this.updateDirection(this.direction === 1 ? -1 : 1);
         } else {
-            this.sorter = sorter || (() => 0);
-            this.sorterChange.emit(this.sorter);
-            this.direction = 1;
-            this.directionChange.emit(1);
+            this.updateSorter(sorter);
+            this.updateDirection(1);
         }
-
-        this.change$.next();
     }
 
     ngAfterViewInit(): void {
         this.changeDetectorRef.detectChanges();
     }
+
+    updateSorter(sorter: TuiComparator<T> | null): void {
+        if (sorter !== this.sorter) {
```

Comment:
If I've not mistaken, this condition is checked twice (some lines above and here).

Do we really need this double check ?

Reply:
oh this function also called independently, from `Sortable` [directive](https://github.com/Tinkoff/taiga-ui/blob/335455cd644d15dad156e6fbe51e3d7246257325/projects/addon-table/components/table/directives/sortable.directive.ts#L36)
But you are right, `Sortable` has the same check, so I removed 👍